Newsletter May 2023 - ready for review (#216)
readthedocs/blog: newsletter-may-2023.rst

.. post:: May 1, 2023
   :tags: newsletter, python
   :author: Ben
   :location: MLM
   :category: Newsletter

.. meta::
   :description lang=en:
      Company updates and new features from the last month,
      current focus, and upcoming features.

Read the Docs newsletter - May 2023
=====================================

News and updates
----------------

- 🚁️ The proxy application El Proxito that serves all of our documentation websites has a new version that's being slowly rolled out.
  Several projects are now served using its new URL resolver,
  which is faster and is receiving new features.

>   We expect to roll it out on all projects soon.

Author: @stsewd can you help with suggestions or inputs for this item? Is there a specific new feature or optimization that should be mentioned?

Author: Custom URLs will be exciting to announce, but I think maybe this announcement is becoming too unspecific then. I'll move it into upcoming work :+1:

Author: I noticed afterwards that I had coupled this announcement with the announcement of improved 404s :see_no_evil: 

So it stays, but I made an extra effort to make the whole update easier to understand.

.. post:: May 1, 2023
   :tags: newsletter, python
   :author: Ben
   :location: MLM
   :category: Newsletter

.. meta::
   :description lang=en:
      Company updates and new features from the last month,

>       current focus, and upcoming features.

Reviewer: This is a more general note but I assume that a lot of newsletters have the exact same meta description. Duplicate meta descriptions are mostly ignored by search engines. We should either customize it a bit -- even if you just add the month and year -- or just remove it.

Author: Removed!

Read the Docs newsletter - May 2023
===================================

News and updates
----------------

- 🚁️ The proxy application El Proxito has been rewritten and the new version is rolled out to selected projects.
  The application resolves URLs for all documentation websites hosted on Read the Docs.
  The new rewrite improves the performance of the resolver and makes it possible to add planned features.
  We are implementing these changes gradually,
  while monitoring their stability.
- 🔎️ ...One of the new features that have been added,
  is an improved 404 page,
  which contains better error messages and tips for users and project owners.

>   It has been rolled out to projects that has the new Proxito resolver.

Reviewer 2: ```suggestion
  It is available to projects using the new Proxito implementation.
```

- 💫️ We now support multiple ``.readthedocs.yaml`` files in the same repository.
  This is especially useful for *monorepos* containing multiple documentation projects with different configurations.
  This allows for instance configurations of several projects with different documentation tools and build environments.
  Read more about the feature :doc:`in our docs <readthedocs:guides/setup/monorepo>`.
- ⚙️ If you use ``build.commands`` in ``.readthedocs.yaml``,
  you are no longer required to have a ``build.tools`` section.
  We changed the validation for ``.readthedocs.yaml`` to accommodate projects that do not need any of the built-in tools exposed.
- 🐛️ Fixed: URLs on pull request builds were pointing to the build page,
  rather than the documentation preview.
  If a build is successful,
  the URL now points to the documentation preview.
- 🔒️ Vulnerability fixed: `CAS session hijacking on Read the Docs for Business <https://github.com/readthedocs/readthedocs.org/security/advisories/GHSA-4mgr-vrh5-hj8q>`__

Upcoming features
-----------------

> - 🚢️ We have concluded work on the version 1.0 of our new `readthedocs-client <https://github.com/readthedocs/readthedocs-client>`_.

Reviewer 2: ```suggestion
- 🚢️ We have concluded work on the first phase of our new `readthedocs-client <https://github.com/readthedocs/readthedocs-client>`_.
```

1.0 feels like it's a release, but we don't have a release yet.

  It's a universal client that enhances documentation for all documentation projects hosted on Read the Docs.
  We will start to use it on a few selected projects.
  If you are interested in having it enabled on your project,
  please reach out.
- 🚁️ As mentioned earlier,
  we have rewritten our proxy application El Proxito and will start to add new features.
  One of the features being worked on is the ability to serve :doc:`subprojects <readthedocs:subprojects>` at custom URL paths.

Want to follow along with our development progress? `View our full Roadmap 📍️`_

.. _View our full Roadmap 📍️: https://github.com/orgs/readthedocs/projects/156/views/1

.. Possible issues

> .. ---------------

Reviewer 2: Maybe mention the problems we had with the urllib3 update?

Author: Good spot, that should happen now - it was an event that took place after the newsletter was finished. We should probably also mention that we are sunsetting the old default images without being too specific (but can reference @humitos's GitHub comment)

Author: It's mentioned now, do you have a second to look? :pray:

- 💫️ We now support multiple ``.readthedocs.yaml`` files in the same repository.
  This is great for monorepos that have several documentation projects with different tooling.
  Read more about the feature :doc:`in our docs <readthedocs:guides/setup/monorepo>`.
- ⚙️ If you use ``build.commands`` in ``.readthedocs.yaml``,
  you are no longer required to have a ``build.tools`` section.
  We changed the validation for ``.readthedocs.yaml`` to accommodate projects that do not need any of the built-in tools exposed.
- 🐛️ Fixed: URLs on pull request builds were pointing to the build page,
  rather than the documentation preview.
  If a build is successful,
  the URL now points to the documentation preview.
- 🔒️ GHSA-4mgr-vrh5-hj8q ...

Upcoming features
-----------------

- 🚢️ We have concluded work on the version 1.0 of our new `readthedocs-client <https://github.com/readthedocs/readthedocs-client>`_.
  It's a universal client that enhances documentation for all documentation projects hosted on Read the Docs.
  We will start to use it on a few selected projects.
  If you are interested in having it enabled on your project,

>   please reach out.

Author: This text was a very rough take. I think we should definitely start mentioning the client, but I'm not sure what the framing should be.

@ericholscher if you have a quick idea here, please LMK :+1: The newsletter isn't ready for review yet but I like that we have this on "important" update in place, it gives the newsletter issue some raison d'etre :)

Reviewer 2: I haven't been following the project that closely, but my feeling is that isn't that ready to start promoting it that widely, maybe @agjohnson can have a more accurate opinion.

Reviewer: Should "reach out" link to https://readthedocs.org/support/ or similar?

Reviewer: Also, I'd sell this a bit more. "It's a universal client that enhances documentation for all documentation projects hosted on Read the Docs." is a fine intro, but it doesn't actually say what it does. Maybe something like:

    The client will be a single library for all the ways Read the Docs enhances documentation from having a flyout menu
    to link to other versions, translations, or subprojects, to showing version warnings for folks browsing outdated docs
    to running ads on community projects.
    There will be integration points and APIs to customize all of these features in the new client.

Reviewer 3: > I haven't been following the project that closely, but my feeling is that isn't that ready to start promoting it that widely

Yeah, this isn't quite that ready yet, it's very beta currently and we are still working out a lot of ops and code patterns still. I'd say this is still alpha level.

> Also, I'd sell this a bit more.

Yeah, agreed here. Related, we are calling these features "addons" instead of "client", particularly for this reason. Though it's not incorrect to call this a client, the function it serves is to enable features in your documentation

Author: It sounds like I should remove it then :+1: 

I'll return with the same request for the next newsletter :) I think it's great to build up some hype here. Especially, I think it's a shame that a lot of team resources are going into "invisible" work, meaning that the newsletter can look very quiet when we're not talking about what we are doing.

Author: I included the suggestions, @davidfischer, thanks - it's going into the June Newsletter draft :+1:

  the URL now points to the documentation preview.
- 🐛️ Fixed: `An issue <https://github.com/readthedocs/readthedocs.org/issues/10290#issuecomment-1542841524>`__ in our legacy build images caused builds using Sphinx to start failing on May 4th when urllib3 2.0.2 was released.
  The issue has been fixed by upgrading OpenSSL on these images.

  We still encourage to specify a newer build image,
  an example is given `in this comment <https://github.com/readthedocs/readthedocs.org/issues/10290#issuecomment-1535120995>`__.
- 🔒️ Vulnerability fixed: `CAS session hijacking on Read the Docs for Business <https://github.com/readthedocs/readthedocs.org/security/advisories/GHSA-4mgr-vrh5-hj8q>`__

Upcoming features
-----------------

- 🚁️ As mentioned earlier,
  we have rewritten our proxy application El Proxito and will start to add new features.
  One of the features being worked on is the ability to serve :doc:`subprojects <readthedocs:subprojects>` at custom URL paths.

.. figure:: img/screenshot-404-juxtaposed.png
   :alt: A screenshot of the old and new versions of our 404 pages.
   
   Left: the classic old 404 page with the ascii art maze - right: new contextualized 404 pages.

>    We are slowly rolling out the new default 404 pages that are meant to be more helpful to users and project owners.

Reviewer 4: I think this image should go where we talk about the 404 improvements instead here, that we are talking about subprojects at custom URL paths
